proto/src/determined/task/v1/task.proto

```diff
@@ -22,6 +22,8 @@ enum State {
   STATE_TERMINATED = 6;
   // The task has begun to exit.
   STATE_TERMINATING = 7;
+  // Additional state to cover queueing operations.
+  STATE_QUEUED = 8;
```

Comment:
with this change, we'll have obsoleted some other states (the api will never return them). I think they should be removed for clarity.

Reply:
would it be reasonable to deprecate states and leave removal for a set future date, in case the user's server version / CLI version / browser version are not all being kept up-to-date

Reply:
i think i changed my mind here, just go ahead and remove them. if their CLI is not up to date (the most likely thing to happen) they will get a warning anyway and it is in the release notes.